doc/self-hosted/build.md

```diff
@@ -0,0 +1,60 @@
+# Build instructions for code-server 🏗️
+
+code-server has several yarn tasks for building the entire binary or specific portions of the app. This guide provides steps to do the aforementioned tasks. Coder recommends having Node 10.15.1+ installed.
```

Comment:
I'd make this a `Requirements` section.

Currently, I think the only requirement is node 10.15.1